packages/compass-e2e-tests/tests/atlas-login.test.ts

```diff
@@ -165,13 +165,28 @@ describe('Atlas Login', function () {
 
       const acceptTOSToggle = browser.$(Selectors.AcceptTOSToggle);
 
-      expect(await acceptTOSToggle.getAttribute('aria-checked')).to.eq('false');
+      expect(await acceptTOSToggle.getAttribute('aria-checked')).to.eq(
+        'false',
+        'Expected TOS toggle to be unchecked'
+      );
 
       await browser.clickVisible(acceptTOSToggle);
 
       await browser.clickVisible(Selectors.AgreeAndContinueButton);
 
-      expect(await acceptTOSToggle.getAttribute('aria-checked')).to.eq('true');
+      await browser.waitUntil(
+        async () => {
+          try {
+            expect(await acceptTOSToggle.getAttribute('aria-checked')).to.eq(
+              'true'
+            );
+            return true;
+          } catch {
+            return false;
+          }
```

Comment:
This is the same, right?

```suggestion
          return await acceptTOSToggle.getAttribute('aria-checked')) === 'true';
```

Reply:
I wanted to keep the assertion to make it clear that we are not just waiting for something, but asserting it. Don't have a strong preference though, will update

Reply:
And pushed again

Reply:
Yeah, I understand, but I also get why (almost?) all of our other `waitUntil` usage is using the return-a-boolean-directly approach, might be worth keeping it consistent?

Reply:
Totally, will keep it consistent